packages/react/primitive/src/Primitive.tsx

```diff
@@ -97,6 +97,10 @@ const AS_ERROR = `Warning: The \`as\` prop has been removed in favour of \`asChi
  *
  * dispatching a custom type within a `discrete` event 👍
  * onPointerDown={(event) => dispatchDiscreteCustomEvent(event.target, new CustomEvent(‘customType’))}
+ *
+ * Note: though React classifies `focus`, `focusin` and `focusout` events as `discrete`, it's  not recommended to use
+ * this utility with them. This is because it's possible for those handlers to be called implicitly during render
+ * e.g. when focus is within a component as it is unmounted, or when managing focus on mount.
```

Comment:
I updated this comment and removed others as they should be covered by this now.